pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/persistent/PersistentReplicator.java

```diff
@@ -617,14 +617,17 @@ public void deleteFailed(ManagedLedgerException exception, Object ctx) {
             return disconnectFuture;
         }
 
-        if (producer != null && (STATE_UPDATER.compareAndSet(this, State.Starting, State.Stopping)
-                || STATE_UPDATER.compareAndSet(this, State.Started, State.Stopping))) {
+        if (producer == null) {
+            // If there's already a reconnection happening, signal to close it whenever it's ready
+            STATE_UPDATER.compareAndSet(this, State.Stopped, State.Stopped);
```

Comment:
here essentially, we are setting the state `Stopped` if it's already `Stopped`. I don't think we need this here.